vortex-serde/src/layouts/read/filtering.rs

```diff
@@ -21,29 +21,31 @@ pub struct RowFilter {
 
 impl RowFilter {
     pub fn new(expr: Arc<dyn VortexExpr>) -> Self {
-        let conjunction = split_conjunction(&expr)
-            .into_iter()
-            .filter(expr_is_filter)
-            .collect();
-
+        let conjunction = split_conjunction(&expr);
         Self { conjunction }
     }
 
     /// Evaluate the underlying filter against a target array, returning a boolean mask
     pub fn evaluate(&self, target: &Array) -> VortexResult<Array> {
-        let mut mask = BoolArray::from(vec![true; target.len()]).into_array();
-        for expr in self.conjunction.iter() {
-            let new_mask = expr.evaluate(target)?;
-            mask = and(new_mask, mask)?;
-
+        let mut filter_iter = self.conjunction.iter();
+        let mut mask = filter_iter
+            .next()
+            .vortex_expect("must have at least one predicate")
+            .evaluate(target)?;
+        for expr in filter_iter {
             if mask.statistics().compute_true_count().unwrap_or_default() == 0 {
-                return Ok(
-                    BoolArray::from_vec(vec![false; target.len()], Validity::AllValid).into_array(),
-                );
+                return BoolArray::try_new(
```

Comment:
ConstantArray?

Reply:
eh, can. There's a bunch of methods missing but can revert back if that turns out to be an issue. I defienitely had a problem where chunked arrays didn't have all methods implemented see #1014